spec/app/domain/authentication/authn-azure/validate_application_identity_spec.rb

```diff
@@ -0,0 +1,428 @@
+            resource_class:             resource_class,
+            validate_azure_annotations: validate_azure_annotations,
+            ).call(
+            account:               account,
+            service_id:            service_id,
+            username:              username,
+            xms_mirid_token_field: xms_mirid_token_field_system_assigned_identity,
+            oid_token_field:       oid_token_field
+          )
+        end
+        it "raises an error" do
+          expect { subject }.to raise_error(::Errors::Authentication::AuthnAzure::RoleMissingConstraint)
+        end
+      end
+
+      context "that has invalid constraint combination in annotations" do
+        before(:each) do
+          allow(host).to receive(:annotations)
+                           .and_return([subscription_id_annotation, resource_group_annotation, user_assigned_identity_annotation, system_assigned_identity_annotation])
+        end
+        subject do
+          Authentication::AuthnAzure::ValidateApplicationIdentity.new(
+            resource_class:             resource_class,
+            validate_azure_annotations: validate_azure_annotations,
+            ).call(
+            account:               account,
+            service_id:            service_id,
+            username:              username,
+            xms_mirid_token_field: xms_mirid_token_field_system_assigned_identity,
+            oid_token_field:       oid_token_field
+          )
+        end
+
+        it "raise an error" do
+          expect { subject }.to raise_error(::Errors::Authentication::IllegalConstraintCombinations)
+        end
+      end
+
+      context "that has invalid Azure annotations" do
+        before(:each) do
+          allow(host).to receive(:annotations)
+                           .and_return([subscription_id_annotation, resource_group_annotation, system_assigned_identity_annotation])
+
+          allow(validate_azure_annotations).to receive(:call)
+                                                 .and_raise('FAKE_VALIDATE_APPLICATION_IDENTITY_ERROR')
+        end
+        subject do
+          Authentication::AuthnAzure::ValidateApplicationIdentity.new(
+            resource_class:             resource_class,
+            validate_azure_annotations: validate_azure_annotations,
+            ).call(
+            account:               account,
+            service_id:            service_id,
+            username:              username,
+            xms_mirid_token_field: xms_mirid_token_field_system_assigned_identity,
+            oid_token_field:       oid_token_field
+          )
+        end
+
+        it "raises an error" do
+          expect { subject }.to raise_error(
+                                  /FAKE_VALIDATE_APPLICATION_IDENTITY_ERROR/
+                                )
+        end
+      end
+    end
+  end
+
+  context "an invalid xms_mirid claim in the Azure token" do
+    context "that does not begin with a /" do
```

Comment:
is this an invalid xms_mirid claim? If so - how come it doesn't raise an error? 
If not - move it to a more proper context

Reply:
moving to be below "A valid xms_mirid claim in the Azure token"
So it becomes:
```
context "A valid xms_mirid claim in the Azure token" do
      context "that does not begin with a /" do
      context "with a valid application identity" do
      context "an invalid application identity" do
```

Is that better placement b/c it relates the xms_mirid? Any farther feels off because the next contexts do not directly have to do with xms_mirid

Reply:
@sigalsax why is this still here?

Reply:
good catch